js/bybit.js

```diff
@@ -2457,284 +2538,317 @@ module.exports = class bybit extends Exchange {
         }, market);
     }
 
-    async fetchOrder (id, symbol = undefined, params = {}) { // TODO
-        if (symbol === undefined) {
-            throw new ArgumentsRequired (this.id + ' fetchOrder() requires a symbol argument');
+    async fetchOrder (id, symbol = undefined, params = {}) {
+        await this.loadMarkets ();
+        let market = undefined;
+        if (symbol !== undefined) {
+            market = this.market (symbol);
+        }
+        let type = undefined;
+        [ type, params ] = this.handleMarketTypeAndParams ('fetchOrder', market, params);
+        if (type !== 'spot' && symbol === undefined) {
+            throw new ArgumentsRequired (this.id + ' fetchOrder() requires a symbol argument for ' + type + ' markets');
         }
+        if (type === 'spot') {
+            // only spot markets have a dedicated endpoint for fetching a order
+            const request = {
+                'orderId': id,
+            };
+            const response = await this.privateGetSpotV1Order (this.extend (params, request));
+            const result = this.safeValue (response, 'result', {});
+            return this.parseOrder (result);
+        }
+        const isUsdcSettled = (market['settle'] === 'USDC');
+        const orderKey = isUsdcSettled ? 'orderId' : 'order_id';
+        params[orderKey] = id;
+        if (isUsdcSettled || market['future'] || market['inverse']) {
+            return this.fetchOpenOrders (symbol, undefined, undefined, params);
```

Comment:
I would suggest that we refrain from emulating fetchOrder with fetchOpenOrders (fetchOrder should return an order by id whether it's open or closed). Or am I missing something?

Reply:
@kroitor You're right but these markets don't have an endpoint that allows doing that, you have always to choose if you want to look in "open" or "closed" orders so in that sense yes this is not 100% correct (if the order is closed won't be discovered), but I've assumed is better to do this instead of throwing a "not supported". Let me know if I should change it

Reply:
> you have always to choose if you want to look in "open" or "closed" orders

It is fine if the exchange has just fetchOpenOrders and fetchClosedOrders for some type of markets.

> but I've assumed is better to do this instead of throwing a "not supported". Let me know if I should change it

I think i'd rather throw an exception in this case, cause the user can always use fetchOpenOrders and/or fetchClosedOrders.

Reply:
@kroitor copy that